### PR TITLE
Achievements Page Mobile View

### LINKED ISSUE #2150: Achievements page mobile view
Currently, the achievements page is not optimized for mobile view.

### PR DESCRIPTION
**Pull request by JovenSoh**
### Description

Closes #2150

Reorganised the achievements page so that its mobile-friendly. The layout has been adjusted to ensure better readability and interaction on smaller screens. Components are now stacked vertically and wrap properly when the screen size does not allow for horizontal alignment. 

Before: There was no mobile display for the achievements page and elements where displayed for landscape view.
image

After: Elements now stack vertically with each other when reaching mobile breakpoint. 
image

There would be auto scroll for child divs which are larger than their parent divs. 
image

### Type of Change

- [x] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Code quality improvements

### How to Test

1. Navigate to the achievements page on a mobile device or use a browser's mobile view functionality.
2. Verify that all components are stacked vertically and are responsive to changes in viewport size.

Please provide any additional steps or configurations if applicable.

### Checklist

- [x] I have tested this code on various mobile screen sizes.

### CODE REVIEW

File: tsconfig.json
Comment (RichDom2185) [inline tsconfig.json:1]: What's this change for?

**Review by RichDom2185 — CHANGES_REQUESTED** [review]
Please double check the diff and do not commit random changes to the repo, I've pushed some reverts to delete all the miscellaneous files and renamings of various files, I'm just left with one question below:

**Comment by RichDom2185** [comment]
Could you provide a screenshot of the UI in the PR description?

File: tsconfig.json
Comment (JovenSoh) [reply tsconfig.json:1]: I was having errors compiling the frontend code, and could not export sharedb-ace.ts to fix it. However, renaming the files seemed to fix it so I have reverted the change to tsconfig.json.

image


image

File: src/commons/achievement/AchievementManualEditor.tsx [hunk 1]
```diff
@@ -95,53 +95,63 @@ const AchievementManualEditor: React.FC<Props> = props => {
...
-      <h3> </h3>
-      <Button outlined={true} text="Update Goal" onClick={updateGoal} intent="primary" />
-
-      <h3> </h3>
-      <Checkbox
-        checked={viewHidden}
-        label="View Hidden Achievements"
-        onChange={() => changeViewHidden(!viewHidden)}
-      />
+      <div className="achievement-manual-editor">
```
Comment (RichDom2185) [inline src/commons/achievement/AchievementManualEditor.tsx:98]: You are wrapping an `achievement-manual-editor` inside an `achievement-manual-editor`, is this intentional?

**Review by RichDom2185 — COMMENTED** [review]
Hi, thanks for working on this! I just have the one comment below:

File: src/commons/achievement/AchievementManualEditor.tsx (hunk 1 above)
Comment (JovenSoh) [reply src/commons/achievement/AchievementManualEditor.tsx:98]: Nope, I have undid this change in a newer commit.

**Comment by JovenSoh** [comment]
Found another issue while testing the Achievements View, more specifically in the control page. The preview was not formatting the achievements correctly, I'm not too sure if this is also the case in production as I do not have admin access.

image

image


Have edited so now it would reflect what its supposed to look like in the achievements page.

image

image

**Review by RichDom2185 — APPROVED** [review]
LGTM, thanks!

**Pull request merged**